tests/testthat/test-get_source_expressions.R

```diff
@@ -79,3 +79,12 @@ test_that("Multi-byte characters correct columns", {
     expect_equal(pc[[1L]]$col1[4L], pc[[1L]]$col1[2L] + 4L)
   })
 })
+
+test_that("Multi-byte character truncated by parser is ignored", {
+  tmp <- tempfile()
+  on.exit(unlink(tmp), add = TRUE)
+  writeLines("y <- x \U2013 42", tmp)  # Unicode en-dash
+  content <- expect_error(get_source_expressions(tmp), NA)
+  expect_equal(content$error$message, "unexpected input")
+  expect_equal(content$error$column_number, 8L)
```

Comment:
Not sure if possible, but have you tried using `with_content_to_parse()` used in the other tests?

Reply:
I took a look at it, but it seemed that the `with_content_to_parse()` itself would fail with the bug. And as it doesn't use the `expect` family of functions internally, the error would not be well formatted. I didn't verify it though.

Reply:
Did some experiments with the tests just now. The error generated by `with_content_to_parse()` ends well formatted anyway by `testthat`. So in principle we could use it, but it doesn't expose the `error` item from `get_source_expressions()` result. I'm sending an alternative version of the tests with a modification to that helper function.